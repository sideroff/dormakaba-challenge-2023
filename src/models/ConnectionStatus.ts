export const ConnectionStatus = {
  Online: 'online',
  Offline: 'offline',
} as const;

export type ConnectionStatus =
  (typeof ConnectionStatus)[keyof typeof ConnectionStatus];
