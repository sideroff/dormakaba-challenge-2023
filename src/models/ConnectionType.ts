export const ConnectionType = {
  Wired: 'wired',
  Wireless: 'wireless',
} as const;

export type ConnectionType =
  (typeof ConnectionType)[keyof typeof ConnectionType];
