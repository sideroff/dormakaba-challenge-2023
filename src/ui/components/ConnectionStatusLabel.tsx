import { ConnectionStatus } from '@/models/ConnectionStatus';
import { Typography } from '@mui/material';
import React from 'react';

interface Props {
  connectionStatus: ConnectionStatus;
}

const ConnectionStatusLabel: React.FC<Props> = ({ connectionStatus }) => {
  const isOnline = connectionStatus === ConnectionStatus.Online;

  return (
    <Typography color={isOnline ? 'success.main' : 'warning.main'}>
      {connectionStatus}
    </Typography>
  );
};

export default ConnectionStatusLabel;
