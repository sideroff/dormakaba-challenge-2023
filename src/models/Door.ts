import { ConnectionType } from './ConnectionType';
import { ConnectionStatus } from './ConnectionStatus';

export interface Door {
  id: string;
  name: string;
  buildingName: string;
  connectionType: ConnectionType;
  connectionStatus: ConnectionStatus;
  lastConnectionStatusUpdate: string;
}
