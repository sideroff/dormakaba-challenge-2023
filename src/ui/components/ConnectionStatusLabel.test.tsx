import { render } from '@testing-library/react';
import ConnectionStatusLabel from './ConnectionStatusLabel';
import { ConnectionStatus } from '@/models/ConnectionStatus';

describe('ConnectionStatusLabel', () => {
  describe('with status `online`', () => {
    it('should render correctly ', () => {
      const { container } = render(
        <ConnectionStatusLabel connectionStatus={ConnectionStatus.Online} />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('with status `offline`', () => {
    it('should render correctly ', () => {
      const { container } = render(
        <ConnectionStatusLabel connectionStatus={ConnectionStatus.Online} />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
