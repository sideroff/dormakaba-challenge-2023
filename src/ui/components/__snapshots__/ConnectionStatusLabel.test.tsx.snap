// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionStatusLabel with status \`offline\` should render correctly  1`] = `
<p
  class="MuiTypography-root MuiTypography-body1 css-18todbp-MuiTypography-root"
>
  online
</p>
`;

exports[`ConnectionStatusLabel with status \`online\` should render correctly  1`] = `
<p
  class="MuiTypography-root MuiTypography-body1 css-18todbp-MuiTypography-root"
>
  online
</p>
`;
